File: ServiceDemo/src/main/java/com/servicedemo/server/BinderPoolService.java
package com.servicedemo.server;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.os.RemoteException;

public class BinderPoolService extends Service {
    public BinderPoolService() {
    }

    @Override
    public IBinder onBind(Intent intent) {
        return new IBinderPool.Stub() {
            @Override
            public IBinder getBinder(int type) throws RemoteException {
                switch (type) {
                    case 1:
                        return new BinderAImpl();
                    default:
                        return new BinderBImpl();
                }
            }
        };
    }

}

File: service-demo/src/main/java/com/servicedemo/server/BinderPoolService.kt
package com.servicedemo.server

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.os.RemoteException
import com.servicedemo.server.IBinderPool.Stub

class BinderPoolService : Service() {
    override fun onBind(intent: Intent): IBinder? {
        return object : Stub() {
            @Throws(RemoteException::class)
            override fun getBinder(type: Int): IBinder {
                return when (type) {
                    1 -> BinderAImpl()
                    else -> BinderBImpl()
                }
            }
        }
    }
}